pkgs/development/python-modules/citeproc-py/default.nix

```diff
@@ -2,39 +2,41 @@
   lib,
   buildPythonPackage,
   fetchPypi,
-  nose,
+  pytestCheckHook,
   git,
   lxml,
   rnc2rng,
+  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "citeproc-py";
   version = "0.6.0";
-  format = "setuptools";
+  pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "d9e3a224f936fe2e5033b5d9ffdacab769cedb61d96c4e0cf2f0b488f1d24b4e";
+    hash = "sha256-2eOiJPk2/i5QM7XZ/9rKt2nO22HZbE4M8vC0iPHSS04=";
   };
 
   buildInputs = [ rnc2rng ];
 
-  propagatedBuildInputs = [ lxml ];
+  build-system = [ setuptools ];
+
+  dependencies = [ lxml ];
 
   nativeCheckInputs = [
-    nose
+    pytestCheckHook
     git
   ];
-  checkPhase = "nosetests tests";
-  doCheck = false; # seems to want a Git repository, but fetchgit with leaveDotGit also fails
+  # doCheck = false;  # seems to want a Git repository, but fetchgit with leaveDotGit also fails
   pythonImportsCheck = [ "citeproc" ];
 
-  meta = with lib; {
```

Comment:
just curious, is this the new best practice?

Reply:
Yes, see #208242.

Reply:
I'm kind of aligned with https://github.com/NixOS/nixpkgs/issues/208242#issuecomment-1700396671. File-wide with usage is an anti pattern, but scoped to something like a meta block is probably harmless. Either way, don't have a strong preference as long we don't flip flop back and forth.

Reply:
I've done it in a bunch of previous PRs and no one's been opposed to it, plus there's no reason to change it back since while not needed here, this is technically better. :shrug: